### PR TITLE
Create API for working with custom forms

### PR DESCRIPTION
**Pull request by francbartoli**
This PR allow people to use custom forms from the client side. An example is the Android [GeoAvalanche app](https://play.google.com/store/apps/details?id=com.geoavalanche.android.app&hl=it) that is currently using such kind of feature

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -9,7 +9,11 @@
 	<property name="bin" location="bin" />
 	<property name="doc" location="doc" />
 	<property name="dist" location="dist" />
+<<<<<<< HEAD
```
Comment (eyedol) [inline build.xml]: Oops, you left some annotations left by the git merge tools

File: src/main/java/com/ushahidi/java/sdk/api/CustomForm.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	
+
+	/* (non-Javadoc)
+	 * @see java.lang.Object#equals(java.lang.Object)
+	 */
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
```
Comment (eyedol) [inline src/main/java/com/ushahidi/java/sdk/api/CustomForm.java]: nitpick: Could we make this consistent with the rest of the code. We go with 

```java
if ( this == obj ) {
    return true;
}

File: src/main/java/com/ushahidi/java/sdk/api/tasks/IncidentsTask.java [hunk 3]
```diff
@@ -152,7 +152,9 @@ public Incidents reportId(int id) {
 
         client.setRequestParameters("by", "incidentid");
         client.setRequestParameters("id", String.valueOf(id));
-        reports = fromString(client.sendGetRequest(url), Reports.class);
+        String response = client.sendGetRequest(url);
+        System.out.println(response);
```
Comment (eyedol) [inline src/main/java/com/ushahidi/java/sdk/api/tasks/IncidentsTask.java]: It seems you left some debugging info

File: src/main/java/com/ushahidi/java/sdk/util/BooleanDeserializer.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.ushahidi.java.sdk.util;
+
+import java.lang.reflect.Type;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+
+public class BooleanDeserializer implements JsonDeserializer<Boolean> {
```
Comment (eyedol) [inline src/main/java/com/ushahidi/java/sdk/util/BooleanDeserializer.java]: Neat

**Comment by eyedol** [comment]
@francbartoli sorry for the late response. Just got back from PTO. I have added some comments for some minor fixes.

**Comment by francbartoli** [comment]
Agreed with your comments. Can you manage the changes alone?

Il giorno 10/ago/2015, alle ore 15:25, Henry Addo <notifications@github.com> ha scritto:

**Comment by eyedol** [comment]
@francbartoli Could you give me more info. What do you mean?

**Comment by francbartoli** [comment]
I’m wondering if you are able to making the changes suggested or you are expecting them from our side and then resubmitting the PR?

Il giorno 10/ago/2015, alle ore 16:51, Henry Addo <notifications@github.com> ha scritto:

**Comment by eyedol** [comment]
Yeah expecting the changes from your end so when I merge, it's still your work. You'll just have to push the update to your branch and the `PR` will be updated automatically.

**Comment by francbartoli** [comment]
The development has been done by a contractor and I’m not quite sure we have internally such skills at the moment. However I’ll try to have a look at your suggestions if I can manage myself without breaking it.

Il giorno 10/ago/2015, alle ore 17:26, Henry Addo <notifications@github.com> ha scritto:

**Comment by eyedol** [comment]
I see, in that case, I'll clean it up then.

**Pull request merged**